### PR TITLE
PDF Viewer Component 

### LINKED ISSUE #100: Integrate acrobat SDK to annotate resumes
* Ask @cyrusdiego for which library to use
* Create a React Component to easily render 
* For now, implement such that onSave, it downloads to computer (this is default) but once #96 is set up will need to integrate with API

### PR DESCRIPTION
**Pull request by cyrusdiego**
# Overview

Create reusable component used to view and mark up a pdf 

# Changes
- components/pdf/PDFViewer.tsx 
- components/pdf/ViewerSDKClient.ts

# Questions & Notes
- How will we be accessing the PDF? URL or byte array? 
    - imo I think it would be a byte array that the server sends to limit the front-end interacting with any other API 

# Issue tracking
Closes #100 

# Testing & Demo
image

### CODE REVIEW

File: www/public/index.html [hunk 1]
```diff
@@ -10,6 +10,7 @@
         <meta content="minimum-scale=1, initial-scale=1, width=device-width" />
         <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
         <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
+        <script type="text/javascript" src="https://documentcloud.adobe.com/view-sdk/main.js"></script>
```
Comment (calebschoepp) [inline www/public/index.html:13]: No npm package I'm assuming?

File: www/src/components/pdf/PDFViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Grid } from '@material-ui/core';
+import React, { FC } from 'react';
+import { useEffect } from 'react';
+
+import ViewSDKClient from './ViewerSDKClient';
+
+// TODO: Modify to properly pass necessary props to make get resume from api
```
Comment (calebschoepp) [inline www/src/components/pdf/PDFViewer.tsx:7]: I'm not really sure what this TODO is saying?

File: www/src/components/pdf/PDFViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const PDFViewer: FC<PDFViewerProps> = () => {
+    useEffect(() => {
+        const viewSDKClient = new ViewSDKClient();
+        viewSDKClient.ready().then(() => {
+            /* Invoke file preview */
+            viewSDKClient.previewFile('adobe-dc-view', {
+                /* Control the viewer customization. */
+                showAnnotationTools: true,
+                enableFormFilling: true,
+                showLeftHandPanel: false,
```
Comment (calebschoepp) [inline www/src/components/pdf/PDFViewer.tsx:21]: What is in the left hand panel?

File: www/src/components/pdf/ViewerSDKClient.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+/*
+Copyright 2020 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in
+accordance with the terms of the Adobe license agreement accompanying
+it. If you have received this file from a source other than Adobe,
+then your use, modification, or distribution of it requires the prior
+written permission of Adobe.
+*/
```
Comment (calebschoepp) [inline www/src/components/pdf/ViewerSDKClient.ts:12]: Any specific reason this is here? Is it required :rofl:

File: www/src/components/pdf/ViewerSDKClient.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        this.readyPromise = new Promise<void>((resolve) => {
+            if (window.AdobeDC) {
+                console.log(this.readyPromise);
+
+                resolve();
+            } else {
+                console.log('yooo');
+                /* Wait for Adobe Document Services PDF Embed API to be ready */
+                document.addEventListener('adobe_dc_view_sdk.ready', () => {
+                    console.log('about to resolve');
```
Comment (calebschoepp) [inline www/src/components/pdf/ViewerSDKClient.ts:43]: Are these logs just TODOs? What is the plan here?

File: www/src/components/pdf/ViewerSDKClient.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    ready(): Promise<void> {
+        return this.readyPromise;
+    }
+
+    previewFile(divId: string, viewerConfig: any): Promise<void> {
+        const config = {
+            /* Pass your registered client id */
+            clientId: '8c0cd670273d451cbc9b351b11d22318',
```
Comment (calebschoepp) [inline www/src/components/pdf/ViewerSDKClient.ts:58]: Pass in through config?

**Review by calebschoepp — CHANGES_REQUESTED** [review]
Great work. Excited to see this integrated.

File: www/public/index.html (hunk 1 above)
Comment (cyrusdiego) [reply www/public/index.html:13]: Not that I could find for the browser :/

File: www/src/components/pdf/PDFViewer.tsx (hunk 2 above)
Comment (cyrusdiego) [reply www/src/components/pdf/PDFViewer.tsx:7]: Oh yeah so it was just a note to pass in any props necessary to integrate with the api. For instance a user id but I wasn't too sure yet how this component will access the pdf from the backend. I want to do that all in this PR tbh so I think I'll ping you to discuss how the pdf stuff will work.

File: www/src/components/pdf/PDFViewer.tsx (hunk 3 above)
Comment (cyrusdiego) [reply www/src/components/pdf/PDFViewer.tsx:21]: When you open adobe pdf viewer it's like the veritcal bar that shows all the pages in the pdf. I figured it's not necessary cus resumes should just be 1 page

File: www/src/components/pdf/ViewerSDKClient.ts (hunk 4 above)
Comment (cyrusdiego) [reply www/src/components/pdf/ViewerSDKClient.ts:12]: loool I just kept it there for no reason. I kinda wanna keep it there to kinda credit Adobe that we're using their code? Idrk how that works though (i.e. properly credit Adobe for using their code), is it sufficient to remove it and just say like "Code adopted from Adobe example code"?

File: www/src/components/pdf/ViewerSDKClient.ts (hunk 5 above)
Comment (cyrusdiego) [reply www/src/components/pdf/ViewerSDKClient.ts:43]: lmao mb those are remnants from me debugging... will remove

File: www/public/index.html (hunk 1 above)
Comment (armiantos) [reply www/public/index.html:13]: Yea seems like this is the norm (or at least this is what they gave in [the samples](https://github.com/adobe/pdf-embed-api-samples/tree/master/More%20Samples/React%20Samples))

File: www/src/util/config.ts [hunk 7]
```diff
@@ -3,6 +3,7 @@ const AUTH0_CLIENT_ID = process.env.REACT_APP_AUTH0_CLIENT_ID || '';
 const SERVER_ENDPOINT = process.env.REACT_APP_SERVER_ENDPOINT || 'http://localhost:1337';
 const SERVER_AUDIENCE = process.env.REACT_APP_SERVER_AUDIENCE || '';
 const MAILCHIMP_URL = process.env.REACT_APP_MAILCHIMP_URL || '';
+const ADOBE_PDF_ID = process.env.ADOBE_PDF_ID || '';
```
Comment (armiantos) [inline www/src/util/config.ts:6]:
```suggestion
const ADOBE_PDF_ID = process.env.REACT_APP_ADOBE_PDF_ID || '';
```
And can you update the `.env.sample` file in `www` to include `MAILCHIMP_URL` and `ADOBE_PDF_ID`? Thanks!

File: www/src/util/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+function arrayBufferToBase64(buffer: ArrayBuffer): string {
```
Comment (armiantos) [inline www/src/util/helpers.ts:1]: Nit: I think this should be a subfile under a `helpers` folder or remove `default` from `export default`

**Review by armiantos — CHANGES_REQUESTED** [review]
Noting that the ViewerPDFClient is just taken from the [sample code](https://github.com/adobe/pdf-embed-api-samples/blob/master/More%20Samples/React%20Samples/src/samples/ViewSDKClient.js) and it will be updated in a separate PR, everything looks good!

Awesome work man 💯 props to you and your perseverance working with plain JS files from SDKs hahahaha

There's a couple nits that we might want to update in this PR before we get it merged in

File: www/src/components/pdf/PDFViewer.tsx (hunk 3 above)
Comment (calebschoepp) [reply www/src/components/pdf/PDFViewer.tsx:16]: Will scrolling work for 2 page resumes?

File: www/src/components/pdf/ViewerSDKClient.ts (hunk 4 above)
Comment (calebschoepp) [reply www/src/components/pdf/ViewerSDKClient.ts:12]: Eh, probably fine to just keep it there I guess

File: www/src/util/config.ts (hunk 7 above)
Comment (cyrusdiego) [reply www/src/util/config.ts:6]: Good catch on these, made the modifications

File: www/src/util/helpers.ts (hunk 8 above)
Comment (cyrusdiego) [reply www/src/util/helpers.ts:1]: fixed

File: www/src/components/pdf/PDFViewer.tsx (hunk 3 above)
Comment (cyrusdiego) [reply www/src/components/pdf/PDFViewer.tsx:16]: Yeah, just tested with a 2-page pdf, you'll be able to scroll (though there shouldn't be any 2 page resumes)

**Review by armiantos — APPROVED** [review]
LGTM! Awesome work 🍻

File: www/src/components/pdf/PDFViewer.tsx (hunk 3 above)
Comment (calebschoepp) [reply www/src/components/pdf/PDFViewer.tsx:16]: > though there shouldn't be any 2 page resumes

I agree. But you would be surprised how many people have 2 page resumes.

**Review by calebschoepp — APPROVED** [review]
Lgtm

**Pull request merged**